python/dlisio/__init__.py

```diff
@@ -255,18 +222,8 @@ def __getitem__(self, type):
         objects : dict
             all objects of type 'type'
         """
-        if type in self.indexedobjects:
-            return self.indexedobjects[type]
-
-        recs = [rec
-            for rec, t
-            in zip(self.attic, self.record_types)
-            if t == type
-        ]
-
-        self.load(recs, reload=False)
-
-        return self.indexedobjects[type]
+        objs = self.metadata.match(type, core.exactmatcher())
+        return { x.fingerprint : x for x in self.promote(objs) }
```

Comment:
You promised to change it to something akin to
```
        objs = self.match('.*', type, core.exactmatcher())?
        return { x.fingerprint : x for x in objs }
```
If you want to use two-arguments match, that means we need to do something with design and implementation of "match" function to allow empty pattern.